### DISCUSSION TITLE
polish crafted-startup for day-to-day use

### DISCUSSION THREAD

**Discussion by erikLundstedt** [SystemCrafters/crafted-emacs#219] [Ideas]
crafted startup feels like a usefull addition but at the moment, a lot of the screen is taken up to info on how to turn it off
what do you think of modifying it(im working on a version myself) to be more like the doom/spacemacs startscreen, removing,or adding a way to disable, the default info on how to turn it off as that is only really relevant when using the default config in my opinion.

it would allso be good to have it be a bit more easily customizable as im currently having to modify the module directly rather than, for example, adding premade,and/or, custom functions to a list in the config file.
for example: I have manually implemented a button(link/menu-entry/whatever) to close the buffer

below is what ive done this far, its still heavily in-dev but it should work good enough

![maim-window-20220809-122230](https://user-images.githubusercontent.com/36701667/183625944-de334078-2fca-4768-bc8e-b438e240451c.png)

**Comment by jeffbowman** [comment]
When I wrote `crafted-startup` my initial thought was to replace the default Emacs splash screen.  Knowing that almost everybody everywhere immediately disables that screen, I didn't want to be "that guy" who turned on a different splash screen which people would be frustrated with and immediately want to turn off.  That is why the instructions on turning it off are provided.  It also mirrors how the default Emacs splash screen looks (there is stuff to remove that screen displayed as well). 

So, I'm glad it is useful to at least some people :-).  

As for extensibility (ie hook functions or similar) to add your own sections, I wasn't sure if this would be generally accepted, so I didn't build anything in, but an API to add modules would certainly be nice to have. 

I make heavy use of the built-in `fancy-splash-*` stuff and plagiarized liberally from the existing Emacs splash screen.  It would be nice to refactor this (significantly) to provide a nice way to override/replace/remove blobs of stuff on the screen. I would envision a list of functions, called in order, which would populate the screen with the "blobs of stuff".

Regardless, I'd be happy to work on this with those who would find the work interesting.

**Reply by erikLundstedt** [reply]
Thanks for the thoughts

One thing we could do is to make the information on disabling it something that is easy to remove by simply removing or disabling the module (blob)



What do you think about the way I have it?


Still need to make the title look a bit nicer on my system

**Reply by erikLundstedt** [reply]
> Thanks for the thoughts
> 
> One thing we could do is to make the information on disabling it something that is easy to remove by simply removing or disabling the module (blob)
> 
> 
> 
> What do you think about the way I have it?
> 
> 
> Still need to make the title look a bit nicer on my system 

Seems I did something that made it not be centered anymore (the title)

**Comment by jeffbowman** [comment]
I prefer the `To Start` section to be horizontal to allow for more screen real estate for other blobs. If that list gets much longer though, a vertical layout would be better to avoid wrapping at the edges.  The text I have on the screen to turn it off was just an idea I had, but the `Dismiss this startup screen` with the checkbox after it was a direct clone from the Emacs startup screen.  I might still like to see the checkbox to `Never show it again` next to the words to dismiss the screen, but that configuration can also be find when following the link to `Customize Crafted Emacs`.  

Overall, the layout is fine.  I'm more interested in what other "blobs" we might put there and how we might enable that customization to this screen.

**Reply by erikLundstedt** [reply]
Might be the fact that I'm coming from using doom (which has a startscreen built-in)to using the spacemacs package (I think that's what I used for a bit)

Booth have the "to get started" as a list like how I did it

**Comment by jeffbowman** [comment]
Gotcha.  Well, ultimately, we can leave that decision up to the user if we go forward with some sort of hook/function list/whatever to customize what is on the screen. I'd leave the "header" and "footer" immutable, but allow the middle pieces to change - ie, the recent files and the "get started" blob. Maybe someone would prefer a list of projects instead of recent files, and maybe they feel the "get started" blob isn't as helpful... whatever... So, if we approach the problem from an API and example implementation, it could be whatever we want (or whatever the user wants actually :-) )
